main_test.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"net/http"
+	"testing"
+)
+
+func TestIsUrl(t *testing.T) {
+ 	tests := []struct {
+ 		name string
+ 		url  string
+ 		want bool
+ 	}{
+ 		{"Valid URL", "http://example.com", true},
+ 		{"Invalid URL", "invalid", false},
+ 		{"Empty string", "", false},
+ 		{"No scheme", "example.com", false},
+ 		{"Empty host", "http://", false},
+ 		{"Invalid host", "http://invalid_host", false},
+ 	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IsUrl(tt.url); got != tt.want {
+				t.Errorf("IsUrl() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestInjectHeaders(t *testing.T) {
+	req, _ := http.NewRequest("GET", "http://example.com", nil)
+	injectHeaders(req)
+
+	if req.Header.Get("Authorization") != "Bearer "+cfg.ServiceAccountToken {
+		t.Errorf("Authorization header not correctly set")
+	}
+
+	tests := []struct {
+		name string
+		username string
+		groups []string
+	}{
+		{"Valid User and Group", "user1", []string{"group1", "group2"}},
+		{"Empty User", "", []string{"group1", "group2"}},
+		{"Empty Group", "user1", []string{}},
+		{"Empty User and Group", "", []string{}},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg.HeaderUsername = tt.username
+			cfg.HeaderGroups = tt.groups
+			injectHeaders(req)
+
+			if req.Header.Get("Impersonate-User") != cfg.HeaderUsername {
```

Comment:
```suggestion
			if req.Header.Get("Impersonate-User") != cfg.HeaderUsername {
```
req.Header.Get("Impersonate-User") should be compared to the value of previous header name with the content of cfg.HeaderUsername. This is also true for cfg.HeaderGroups.  These values are the headers names, not the content